roachtest: add some shared-process multi-tenant configs

yuzefovich opened the pull request:
This PR adds shared-process multi-tenant configs to `multitenant/tpch`, `tpchvec`, `tpccbench`, `kv0`, and `kv95` roachtests. The comparison between single-tenant and shared-process multi-tenant shows that there is no significant difference on `multitenant/tpch` and `tpchvec` but there is a perf hit of 2-8% on all other tests. See #104379 for more details.

Fixes: #104379.

Release note: None

yuzefovich committed:
roachtest: introduce shared-process multitenant/tpch configs

This commit extends the existing `multitenant/tpch` test to run the
multi-tenant deployment in the shared-process mode (in addition to
already existing separate-process mode).

The comparison of single-tenant vs shared-process multi-tenant (averaged
over 25 runs):
```
Q1:	before: 6.29s	after: 6.29s	 0.06%
Q2:	before: 4.25s	after: 4.22s	 -0.69%
Q3:	before: 4.43s	after: 4.42s	 -0.19%
Q4:	before: 1.45s	after: 1.45s	 -0.25%
Q5:	before: 2.81s	after: 2.85s	 1.39%
Q6:	before: 7.67s	after: 7.60s	 -0.84%
Q7:	before: 7.33s	after: 7.08s	 -3.46%
Q8:	before: 1.11s	after: 1.14s	 2.49%
Q9:	before: 9.03s	after: 9.42s	 4.38%
Q10:	before: 2.00s	after: 1.99s	 -0.60%
Q11:	before: 1.05s	after: 1.08s	 2.35%
Q12:	before: 8.06s	after: 8.09s	 0.37%
Q13:	before: 2.34s	after: 2.35s	 0.29%
Q14:	before: 0.75s	after: 0.75s	 0.05%
Q15:	before: 4.05s	after: 4.07s	 0.43%
Q16:	before: 0.98s	after: 0.98s	 0.82%
Q17:	before: 0.29s	after: 0.28s	 -1.12%
Q18:	before: 4.14s	after: 4.07s	 -1.47%
Q19:	before: 0.53s	after: 0.53s	 1.14%
Q20:	before: 17.75s	after: 17.31s	 -2.49%
Q21:	before: 7.45s	after: 7.05s	 -5.31%
Q22:	before: 0.66s	after: 0.63s	 -3.41%
```

Release note: None

yuzefovich committed:
roachtest: add shared-process MT configs to tpccbench, kv0, and kv95

This commit extends the tpccbench roachtest to also run two configs
(`nodes=3/cpu=16` and `nodes=3/cpu=4`) in the shared-process
multi-tenant mode. It also does the same in four configs in kv0 and kv95
each. The comparison of single-tenant vs shared-process multi-tenant
using these new configs shows that there is a perf hit of using the
latter (on the order of 2-8%, more details are in #104379).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104703)
<!-- Reviewable:end -->

yuzefovich committed:
roachtest: extend tpchvec for shared-process multi-tenant

This commit extends the `tpchvec` roachtest to also run in
shared-process multi-tenant mode. Additionally, it adds another config
that compares the performance with the KV projection pushdown enabled vs
disabled.

Comparison of single-tenant and shared-process multi-tenant shows
effectively no difference (average over 50 runs):
```
Q1:	before: 2.86s	after: 2.95s	 3.09%
Q2:	before: 2.95s	after: 2.95s	 -0.20%
Q3:	before: 2.29s	after: 2.19s	 -4.10%
Q4:	before: 1.56s	after: 1.59s	 1.61%
Q5:	before: 2.35s	after: 2.31s	 -1.92%
Q6:	before: 4.00s	after: 3.97s	 -0.88%
Q7:	before: 5.01s	after: 4.98s	 -0.52%
Q8:	before: 1.01s	after: 1.00s	 -1.03%
Q9:	before: 5.08s	after: 4.81s	 -5.29%
Q10:	before: 1.92s	after: 1.97s	 2.94%
Q11:	before: 0.85s	after: 0.85s	 -0.54%
Q12:	before: 4.10s	after: 3.91s	 -4.49%
Q13:	before: 0.96s	after: 0.99s	 2.80%
Q14:	before: 0.42s	after: 0.42s	 0.24%
Q15:	before: 2.28s	after: 2.25s	 -1.30%
Q16:	before: 0.83s	after: 0.78s	 -5.68%
Q17:	before: 0.22s	after: 0.22s	 0.63%
Q18:	before: 1.81s	after: 1.79s	 -0.86%
Q19:	before: 0.44s	after: 0.44s	 0.96%
Q20:	before: 8.46s	after: 9.52s	 12.55%
Q21:	before: 4.55s	after: 4.39s	 -3.63%
Q22:	before: 0.52s	after: 0.50s	 -3.97%
```

Release note: None

yuzefovich commented:
I reran `tpchvec` with the latest fix to the DistSQL physical planner, and the performance difference went away (details are in the third commit). I updated the PR description accordingly. PTAL.

yuzefovich commented:
TFTR!

bors r+
